CheckerboardKarel.java

```diff
@@ -1,16 +1,107 @@
-/*
- * File: CheckerboardKarel.java
- * ----------------------------
- * When you finish writing it, the CheckerboardKarel class should draw
- * a checkerboard using beepers, as described in Assignment 1.  You
- * should make sure that your program works for all of the sample
- * worlds supplied in the starter folder.
- */
-
-import stanford.karel.*;
-
-public class CheckerboardKarel extends SuperKarel {
-
-	// You fill in this part
-
-}
+/*
+ * File: CheckerboardKarel.java
+ * ----------------------------
+ * When you finish writing it, the CheckerboardKarel class should draw
+ * a checkerboard using beepers, as described in Assignment 1.  You
+ * should make sure that your program works for all of the sample
+ * worlds supplied in the starter folder.
+ */
+
+import stanford.karel.*;
+
+public class CheckerboardKarel extends SuperKarel {
+	public void run() {
+		/*start of main
+		 * Karel drop beeper
+		 */
+		if (beepersPresent() ) {
+			move();
+		}else{
+		putBeeper();
+		}
+		
+		//Karel move in two steps and drop beeper and then move to another row
+		beeperDropping();
+		madeEasy();
+		
+		//Karel move in two steps and drop beeper and then move to another row
+		beeperDropping();
+		move();
+		turnDoubleRight();
+		
+		//Karel move in two steps and drop beeper and then move to another row
+		putBeeper();
+		beeperDropping();
+		
+		//Karel move in two steps and drop beeper and then move to another row
+		madeEasy();
+		beeperDropping();
+		move();
+		turnDoubleRight();
+		
+		//Karel move in two steps and drop beeper and then move to another row
+		putBeeper();
+		beeperDropping();
+		
+		//Karel move in two steps and drop beeper and then move to another row
+		madeEasy();
+		beeperDropping();
+		move();
+		turnDoubleRight();
+		
+		//Karel move in two steps and drop beeper and then move to another row
+		putBeeper();
+		beeperDropping();
+		
+		//Karel move in two steps and drop beeper and then move to another row
+		madeEasy();
+		beeperDropping();
+		move();
+		turnRight();
+		
+		//end of public methods
+	}
+		
+		
+	/*start of private methods
+	 * this section declare my own person methods
+	 * leggo!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
+	 */
+
+	//Karel move in two steps
+	private void moveDouble() {
+		move();
+		move();
+	}
+	//karel make a turn around move in left sideways
+	private void turnDoubleLeft() {
+		turnLeft();
+		move();
+		turnLeft();
+	}
+	//karel make a turn around move in right sideways
+	private void turnDoubleRight() {
+		turnRight();
+		move();
+		turnRight();
+	}
+	//karel is making a double move and also dropping beeper
+	private void beeperDropping() {

```

Comment:
This method name is confusing. The method says beeperDropping, yet in its implementation there are three beepers that are being put. Why is the method name then referring to just one beeper?